public/app/core/utils/richHistory.ts

```diff
@@ -329,6 +329,55 @@ export function notEmptyQuery(query: DataQuery) {
   return false;
 }
 
+export function filterQueriesBySearchFilter(queries: RichHistoryQuery[], searchFilter: string) {
+  const filteredQueries = queries.filter(query => {
+    const searchInComment = query.comment.includes(searchFilter);
+    const searchInQueries = query.queries.filter(query => {
+      const queryHasSearchedValue = Object.values(_.omit(query, 'datasource')).some(value =>
+        value.toString().includes(searchFilter)
+      );
+      return queryHasSearchedValue;
```

Comment:
Storing the result instead of returning it straight away seems a bit excessive here. ;) (Could even do an implicit return.)
The same goes for the other filter functions.

Reply:
Resolved in https://github.com/grafana/grafana/pull/25747/commits/3313c6b0e430ceeb85beb17670c8b1215518229a and https://github.com/grafana/grafana/pull/25747/commits/39a6c845d9891251f7dd11827642fcfd4a690b6b